NKL/KLR.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Govereau
+-/
+
+
+/-!
+# Abstract syntax of Core NKL language
+
+This language is the result of "tracing", and is used as the
+portable format, a.k.a. Kernel Language Representation (KLR).
+-/
+
+namespace NKL.KLR
+
+-- TODO
+inductive Ty where
+
+inductive Const where
+  | none
+  | bool (value : Bool)
+  | int (value : Int)
+  | float (value : Float)
```

Comment:
Will we eventually need Float32/Float16/Float8 etc?

Reply:
I am not sure. These expressions are, conceptually, python expressions. So, really this should be a big float?
It is only when these constants reach a tensor that we need to check if they are compatible.